src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1195,16 +1195,155 @@ private ImplicitNamedTypeSymbol BindScriptClass()
+            if (previousSubmission == null)
+            {
+                return newUsings;
+            }
+
+            var externalUsings = ArrayBuilder<NamespaceOrTypeAndUsingDirective>.GetInstance();
+            var uniqueTargets = PooledHashSet<NamespaceOrTypeSymbol>.GetInstance();
+
+            foreach (var previousUsing in previousSubmission.ExternalUsings)
+            {
+                var previousTarget = previousUsing.NamespaceOrType;
+                if (previousTarget.IsType)
+                {
+                    if (uniqueTargets.Add(previousTarget))
+                    {
+                        externalUsings.Add(previousUsing);
+                    }
+                }
+                else
+                { 
+                    var expandedNamespace = ExpandPreviousSubmissionNamespace((NamespaceSymbol)previousTarget, this.GlobalNamespace);
+                    if (uniqueTargets.Add(expandedNamespace))
+                    {
+                        externalUsings.Add(new NamespaceOrTypeAndUsingDirective(expandedNamespace, previousUsing.UsingDirective));
+                    }
+                }
+            }
+
+            // The previous submission should not have had extern aliases (unsupported)
+            // and using aliases are handled separately (during lookup) so we only need
+            // to extract the regular usings.
+            Debug.Assert(previousSubmission.SubmissionImports.ExternAliases.Length == 0);
+
+            // TODO (https://github.com/dotnet/roslyn/issues/5517): uncomment
+            //if (this.ReferenceManagerEquals(previousSubmission))
+            //{
+            //    foreach (var previousUsing in previousSubmission.SubmissionImports.Usings)
+            //    {
+            //        if (uniqueTargets.Add(previousUsing.NamespaceOrType))
+            //        {
+            //            externalUsings.Add(previousUsing);
+            //        }
+            //    }
+            //}
+            //else
+            {
+                foreach (var previousUsing in previousSubmission.SubmissionImports.Usings)
+                {
+                    var previousTarget = previousUsing.NamespaceOrType;
+                    if (previousTarget.IsType)
+                    {
+                        if (uniqueTargets.Add(previousTarget))
+                        {
+                            externalUsings.Add(previousUsing);
+                        }
+                    }
+                    else
+                    {
+                        var expandedNamespace = ExpandPreviousSubmissionNamespace((NamespaceSymbol)previousTarget, this.GlobalNamespace);
+                        if (uniqueTargets.Add(expandedNamespace))
+                        {
+                            externalUsings.Add(new NamespaceOrTypeAndUsingDirective(expandedNamespace, previousUsing.UsingDirective));
+                        }
+                    }
+                }
```

Comment:
I'm pretty sure they used to be different, but they certainly aren't now.  I've merged them into a single loop using ```Concat```.